frontend/dockerfile/dockerfile2llb/convert.go

```diff
@@ -1130,6 +1134,41 @@ func isReachable(from, to *dispatchState) (ret bool) {
 	return false
 }
 
+func hasCircularDependency(states []*dispatchState) (bool, *dispatchState) {
+       var visit func (state *dispatchState) bool
+       if states == nil {
+               return false, nil
+       }
+       visited := make(map[*dispatchState]struct{})
+       path := make(map[*dispatchState]struct{})
+
+       visit = func (state *dispatchState) bool {
```

Comment:
nit: can be `visit := ...`

Reply:
visit is called recursively, I get "undefined: visit" when using " := "; this is the go's order of evaluation